cpp/src/arrow/parquet/writer.cc

```diff
@@ -143,6 +153,82 @@ Status FileWriter::Close() {
 
 FileWriter::~FileWriter() {}
 
+// Create a slice of a PrimitiveArray.
+//
+// This method is specially crafted for WriteFlatTable and assumes the following:
+//  * chunk_size is a multiple of 512
```

Comment:
it might be good to add a note on the exact lines that make this assumption.  Its not immediately obvious to me.  Also, do consider adding a DCHECK to assert this.